ecms_base/modules/custom/ecms_migration/src/Plugin/migrate/process/RemoveAttribute.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ecms_migration\Plugin\migrate\process;
+
+use Drupal\migrate\MigrateException;
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+
+/**
+ * Uses a regex to remove the given attribute.
+ *
+ * @MigrateProcessPlugin(
+ *   id = "remove_attribute"
+ * )
+ *
+ * Remove 'style' attributes from HTML tags:
+ * @code
+ * field_text:
+ *   plugin: remove_attribute
+ *   source: text
+ *   attribute: 'style'
+ * @endcode
+ */
+class RemoveAttribute extends ProcessPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    if (!isset($this->configuration['attribute'])) {
+      throw new MigrateException('"search" must be configured.');
+    }
+
+    $attributeToRemove = $this->configuration['attribute'];
+    return preg_replace('/(<[^>]+) ' . $attributeToRemove . '=".*?"/i', '$1', $value);
```

Comment:
My regex isn't the best, but will this affect any html that has a style tag in the markup? e.g.:

```
<div>
<style>body {background-color: red;}</style>
</div>
```
Should it also remove those style tags?

Reply:
It won't catch these, it will only catch an inline attribute